### DISCUSSION TITLE
Failing to install openEMS on Virtual Machine

### DISCUSSION THREAD

**Discussion by RFCommander** [thliebig/openEMS-Project#155] [General]
First of all, long time CST user and discovering openEMS made my week.

Windows user, but created a Virtual Machine(linuxmint 21.2 cinnamon) to experiment with openEMS.
After installing everything on the requirements list https://docs.openems.de/install.html#requirements I cloned the github Project as stated in:

git clone --recursive https://github.com/thliebig/openEMS-Project.git
cd openEMS-Project
./update_openEMS.sh ~/opt/openEMS

Running ./update_openEMS.sh freezes my Virtual Machine and it stays there forever until I close forcefully the VM.
Any indication on why this happen?

Thank you in advance

**Comment by biergaizi** [comment]
> Running `./update_openEMS.sh` freezes my Virtual Machine and it stays there forever until I close forcefully the VM.
Any indication on why this happen?

Sounds like a out-of-memory problem. I recommend assigning at least 4 vCPUs and 8 GiB RAM to your VM, otherwise even if the installation can complete, it will have difficulty running simulations.

But if you really do insist on running the VM with low RAM (or if the problem is not out-of-memory but something else and you need to rule that out), then open `./update_openEMS.sh` and make these changes:

```patch
diff --git a/update_openEMS.sh b/update_openEMS.sh
index 86c992f..34c3443 100755
--- a/update_openEMS.sh
+++ b/update_openEMS.sh
@@ -88,7 +88,7 @@ if [ -f configure ]; then
 fi
 
 echo "compiling $1 ... please wait"
-make -j4 >> $LOG_FILE
+make -j1 >> $LOG_FILE
 if [ $? -ne 0 ]; then
   echo "make for $1 failed"
   cd ..
@@ -121,7 +121,7 @@ if [ $? -ne 0 ]; then
   exit 1
 fi
 echo "build openEMS and dependencies ... please wait"
-make -j5 >> $LOG_FILE 2>&1
+make -j1 >> $LOG_FILE 2>&1
 if [ $? -ne 0 ]; then
   echo "make failed, build incomplete, please see logfile for more details..."
   cd $basedir
```

**Reply by RFCommander** [reply]
Thank you for your anwser. Unfortuntelly/fortunatelly I installed openEMS on Windows directly and it worked, after changing how cython is imported (thanks to https://github.com/thliebig/openEMS/issues/127).

I did not verified if your approach on VM works.
Once again, thank you for your help.
